catalog/MDCCatalog/MDCNodeListViewController.swift

```diff
@@ -62,22 +62,19 @@ class MDCNodeListViewController: CBCNodeListViewController {
   override init(node: CBCNode) {
     super.init(node: node)
 
+    var childrenNodes = node.children.filter { $0.isExample() }
+
     // Make sure that primary demo appears first
-    let orderedNodes = NSMutableArray()
-    for childNode in node.children {
-      if childNode.isExample() {
-        let isPrimaryDemo = childNode.isPrimaryDemo()
-        if isPrimaryDemo {
-          orderedNodes.insert(childNode, at: 0)
-          componentDescription = childNode.exampleDescription()
-        } else {
-          orderedNodes.add(childNode)
-        }
-      }
+    if let primaryDemoNodeIndex = childrenNodes.index(where: { $0.isPrimaryDemo() }),
+        primaryDemoNodeIndex != 0 {
+      let primaryDemoNode = childrenNodes[primaryDemoNodeIndex]
+      childrenNodes.remove(at: primaryDemoNodeIndex)
+      childrenNodes.insert(primaryDemoNode, at: 0)
```

Comment:
I saw a one-line way to move elements:
`insert(remove(at: primaryDemoNodeIndex), at: 0)`

Reply:
I find that snippet somewhat too clever, what do you think about the following?

```swift
let primaryDemoNode = childrenNodes.remove(at: primaryDemoNodeIndex)
childrenNodes.insert(primaryDemoNode, at: 0)
```

Reply:
I'll take it. I just wish the language had 'move'